lib/plus_plus/base.rb

```diff
@@ -28,27 +28,19 @@ def plus_plus_on_change(*args)
         dup = self.dup
         changes.each { |k, v| dup[k] = v.first }
 
-        changed = options[:changed]
-        prev_satisfied_for_minus = if options[:minus].respond_to?(:call)
-                                     dup.instance_exec(&options[:minus])
-                                   else
-                                     dup.send(changed) == options[:minus]
-                                   end
-        self_satisfied_for_plus = if options[:plus].respond_to?(:call)
-                                    instance_exec(&options[:plus])
-                                  else
-                                    send(changed) == options[:plus]
-                                  end
-        self_satisfied_for_minus = if options[:minus].respond_to?(:call)
-                                     instance_exec(&options[:minus])
-                                   else
-                                     send(changed) == options[:minus]
-                                   end
-        prev_satisfied_for_plus = if options[:plus].respond_to?(:call)
-                                    dup.instance_exec(&options[:plus])
-                                  else
-                                    dup.send(changed) == options[:plus]
-                                  end
+        prev_satisfied_for_minus = plus_plus_satisfied_for?(dup,
```

Comment:
I think this would be way easier to read (and use less space) when written as 

```
prev_satisfied_for_minus =
  plus_plus_satisfied_for?(dup, options[:minus],  options[:changed])
```